torch/onnx/_internal/exporter/_onnx_program.py

```diff
@@ -109,8 +118,35 @@ def _create_value_mapping(graph: ir.Graph) -> dict[str, ir.Value]:
     return values
 
 
+def _to_ort_value(tensor: torch.Tensor) -> ort.OrtValue:
+    """Convert a PyTorch tensor to an ONNX Runtime OrtValue."""
+    import onnxruntime as ort
+
+    if hasattr(ort.OrtValue, "ortvalue_from_numpy_with_onnx_type"):
+        # This requires ONNX Runtime 1.21 or newer
+        if tensor.dtype in _NP_UNSUPPORTED_DTYPES:
+            onnx_type = _core.torch_dtype_to_onnx_dtype(tensor.dtype)
+            return ort.OrtValue.ortvalue_from_numpy_with_onnx_type(
+                tensor.numpy(force=True), onnx_element_type=onnx_type
+            )
+    # TODO(justinchuby): Use dlpack when ORT properly supports it
+    return ort.OrtValue.ortvalue_from_numpy(tensor.numpy(force=True))
+
+
+def _from_ort_value(value: ort.OrtValue) -> torch.Tensor:
+    if value.element_type() == ir.DataType.BFLOAT16:
+        # This requires ONNX Runtime 1.21 or newer
```

Comment:
Should we do version control for users? After all, ort version is not a required/guard in pytorch.

Reply:
Added